mypy/newsemanal/typeanal.py

```diff
@@ -1281,3 +1286,23 @@ def make_optional_type(t: Type) -> Type:
         return UnionType(items + [NoneTyp()], t.line, t.column)
     else:
         return UnionType([t, NoneTyp()], t.line, t.column)
+
+
+def fix_instance_types(t: Type, fail: Callable[[str, Context], None]) -> Type:
+    """Recursively fix all instance types (type argument count) in a given type.
+
+    For example 'Union[Dict, List[str, int]]' will be transformed into
+    'Union[Dict[Any, Any], List[Any]]'.
+    """
+    return t.accept(InstanceFixer(fail))
+
+
+class InstanceFixer(TypeTranslator):
```

Comment:
I think that you can use `TypeTraverserVisitor` here instead, since you modify types in place. It will be more efficient as it doesn't construct new types.